ruma-federation-api/src/membership/create_join_event/v2.rs

```diff
@@ -0,0 +1,37 @@
+//! [PUT /_matrix/federation/v2/send_join/{roomId}/{eventId}](https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v2-send-join-roomid-eventid)
+
+use ruma_api::ruma_api;
+use ruma_events::pdu::PduStub;
+use ruma_identifiers::{EventId, RoomId};
+
+use super::RoomState;
+
+ruma_api! {
+    metadata: {
+        description: "Send a join event to a resident server.",
+        name: "create_join_event",
+        method: PUT,
+        path: "/_matrix/federation/v2/send_join/:room_id/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The room ID that is about to be joined.
+        #[ruma_api(path)]
+        pub room_id: RoomId,
+        /// The user ID the join event will be for.
+        #[ruma_api(path)]
+        pub event_id: EventId,
+
+        /// PDU type without event and room IDs.
+        #[ruma_api(body)]
+        pub pdu_stub: PduStub,
+    }
+
+    response: {
+        /// Full state of the room.
+        #[serde(with = "crate::serde::room_state")]
```

Comment:
This shouldn't be necessary; that module is used as a workaround to deserialize a weird tuple syntax that accidentally showed up in the spec. (I should probably rename that module to something more descriptive. :thinking:)
```suggestion

```

Reply:
By weird tuple syntax, do you mean the response format in v1 of send join?

Reply:
Ok I found issue #2541 in matrix-org/matrix-doc where this was discussed, so it makes more sense now.